hazelcast/src/main/java/com/hazelcast/cluster/ClusterClock.java

```diff
@@ -20,12 +20,21 @@
 
     /**
      * Returns the cluster-time in milliseconds.
-     *
      * The cluster-time measures elapsed time since the cluster was created. Comparable to the {@link System#nanoTime()}.
-     *
      * @return the cluster-time (elapsed milliseconds since the cluster was created).
      */
     long getClusterTime();
 
     long getClusterTimeDiff();
+
+    /**
+     * Returns the cluster start time in nanoseconds.
+     * When first node in cluster becomes master, its System#nanoTime() value is saved as
+     * clusterStartTime. Although in most JVM, System#nanoTime() is not related to
+     * any other notion of system or wall-clock time, be careful about using this value
+     * in a critical place, this value is only used for statistics purpose for calculating
+     * estimated up-time of  cluster.
+     * @return the cluster-start-time
+     */
+     long getClusterStartTime();
```

Comment:
This may return millis as in your first commit. It may be confusing to return nanos since `long getClusterTime();` is returning millis but `getClusterStartTime ` is returning nanos. And also `RuntimeMXBean#getStartTime` returns result in millis.